### PR TITLE
chore(ci): libsinsp e2e CI caching

### PR DESCRIPTION
**Pull request by therealbobo**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area CI

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

This adds build caching in libsinsp e2e CI jobs.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/e2e_ci.yml [hunk 1]
```diff
@@ -96,7 +88,45 @@ jobs:
...
+    name: test-e2e-${{ matrix.arch }}-${{ matrix.driver.name }} 😇 (bundled_deps)
+    needs: [build-test-e2e]
+    runs-on: ${{ (matrix.arch == 'arm64' && 'actuated-arm64-8cpu-16gb') || 'ubuntu-22.04' }}
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+        driver: [ {name: kmod, option: -k}, {name: bpf, option: -b}, {name: modern-bpf, option: -m} ]
+      fail-fast: false
+    steps:
+      - name: Restore build skeleton
```
Comment (FedeDP) [inline .github/workflows/e2e_ci.yml:112]: Doesn't this mean that we always build tests, then cache the result, and then in `test-e2e` job we restore the just-cached build dir?
I think the proper way is to always try to restore-first (ie: keeping a single job) and if that fails, then build and cache for subsequent runs.

Comment (therealbobo) [reply .github/workflows/e2e_ci.yml:112]: This is just a first step: instead of building 6 times the same executable, this builds only once for arch. :)

**Review by FedeDP — APPROVED** [review]
Love this approach!
Wondering whether we can use it and save some precious CI time elsewhere!
/approve

**Comment by therealbobo** [comment]
Probably yes but we’ll need to check if there is a storage limit for the gh runners (the build dirs are pretty heavy :/)

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**